### ISSUE TITLE
Consuming/processing GPS data

### ISSUE THREAD

**Issue by mmoses1127** [mmoses1127/Bravo#2]
Been reading about GPS data. Does this sound right for my schema?

From what I'm reading, each 'route' in my routes table will have 1000s of 'gps_points', which each have their own table with lat, long, elevation, speed, etc.  

These are all parsed from the GPS data flowing in from the satellites. I imagine there is a particular library for triggering a device to fetch GPS data and then for parsing it.

**Comment by mwmadsen67** [comment]
I think it would be too much work to deal with 1000s of data points for this project, not to mention 1000s of data points that each have a lat, long, elevation and speed. Definitely if you wanted to implement GPS data this way then this would be it's own feature in itself. 

For a starting point, it might be easier to simplify this down to a lat and long for each GPS point. Furthermore, you could store a list of data points as a string, like so:
"[37.808640, -122.396896],[37.812620, -122.396156],[37.817094, -122.395824]"
this would allow you to handle this with just one column.

Down the line if you want more interesting data to play with you could try and explore a GPS points table with more complex info.

**Comment by mwmadsen67** [comment]
Also in the future you dont need to make a new issue for a question like this, you can just ask me in the thread of the other issue

**Comment by mmoses1127** [comment]
Hey Mike,

I think I have found a nice compromise between the need to have GPS data for displaying the ride on a map and the need to keep my schema simplified.

I can include a column in 'rides' that contains strings with URLs pointing to GPX files. I will make this data optional. I can use [GPXParser.js](https://www.npmjs.com/package/gpxparser) to parse the GPX file and display it on my site using Leaflet. This seems like a common solution and should be possible to implement in my timeframe. The guide I referred to is [here](https://www.manuelkruisz.com/blog/posts/gpx-maps-react).

**Comment by mwmadsen67** [comment]
First I'll say I've never worked with GPX files before, so I will be learning how to use them as you do.

I think it makes sense to store some data from the GPX files in your database for easy reference.

I am a little bit concerned about getting bogged down in some of the knitty gritty of getting a separate library to work as part of your first feature. My inclination is to focus on getting basic paths working with simple lat long coordinates, and then later adding the extra GPX data. If, from your own research, you feel that its doable then I'll trust you to get it done. However I would put a cap on the amount of time you spend on just getting GPX data to work (1 - 2 days max), since the other features of your app will take a lot of time on their own.

**Comment by mmoses1127** [comment]
Thanks Mike. Maybe I am misunderstanding, but without a GPX file, you would need to generate lat and long data somehow. And for even the most basic routes, it would be hundreds of points. If that's right, having a GPX with a parser handle all that would actually be the simplest way. What do you think about me building out my "rides" feature without the GPX data initially, but with an image of the map? Then, once I finish this and a couple other features, I can come back around and add the GPX functionality. I think that this GPS rendering on the map is a central feature of the app, so I hope I can get to it.

**Comment by mwmadsen67** [comment]
Yeah the simplest way would be to hardcore some gps points, you would only need a couple points and then just draw a line between them to end up with a route along a road. It wouldn't be nearly as built out as a real route from the app of course.
